tests/modules/services/gpg-agent/default-homedir.nix

```
@@ -2,7 +2,13 @@
 
 with lib;
 
-{
+let
+  inherit (pkgs.stdenv.hostPlatform) isLinux;
+
+  socketPath =
+    mkIf isLinux config.systemd.user.sockets.gpg-agent.Socket.ListenStream;
```

```suggestion
    config.systemd.user.sockets.gpg-agent.Socket.ListenStream;
```

`mkIf` only makes sense within the module system (it simply wraps its value in an attribute set with the condition, see the [definition](https://github.com/NixOS/nixpkgs/blob/b55a4988418b73079e7c50ee574545a9c956ee0b/lib/modules.nix#L912-L915)). Since there is already an `optionalString` around the usage of `socketPath` this is entirely unnecessary.